pkg/agent/flowexporter/connections/connections.go

```diff
@@ -126,3 +132,59 @@ func lookupServiceProtocol(protoID uint8) (corev1.Protocol, error) {
 	}
 	return serviceProto, nil
 }
+
+func (cs *connectionStore) addItemToQueue(connKey flowexporter.ConnectionKey, conn *flowexporter.Connection) {
+	currTime := time.Now()
+	pqItem := &flowexporter.ItemToExpire{
+		Conn:             conn,
+		ActiveExpireTime: currTime.Add(cs.expirePriorityQueue.ActiveFlowTimeout),
+		IdleExpireTime:   currTime.Add(cs.expirePriorityQueue.IdleFlowTimeout),
+	}
+	heap.Push(cs.expirePriorityQueue, pqItem)
+	cs.expirePriorityQueue.KeyToItem[connKey] = pqItem
+}
+
+func (cs *connectionStore) AcquireConnStoreLock() {
+	cs.mutex.Lock()
+}
+
+func (cs *connectionStore) ReleaseConnStoreLock() {
+	cs.mutex.Unlock()
+}
+
+// deleteConnWithoutLock deletes the connection from the connection map given
+// the connection key without grabbing the lock. Caller is expected to grab lock.
+func (cs *connectionStore) deleteConnWithoutLock(connKey flowexporter.ConnectionKey, isDenyConn bool) error {
+	_, exists := cs.connections[connKey]
```

Comment:
why do we need `isDenyConn` as a parameter when we are actually looking up the connection? Isn't this info stored in the connection struct?

Reply:
on second thought, not sure why we don't have separate implementations for `ConntrackConnectionStore` and `DenyConnectionStore`?

Reply:
`isDenyConn` is not stored in the connection struct.
I wanted to avoid code repetition between the two connection stores. 
Agree that having separate implementation might better than having the bool. Changed to separate implementation.